node/pkg/chain/helper/signer.go

```diff
@@ -0,0 +1,227 @@
+package helper
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"math/big"
+	"os"
+	"strings"
+	"time"
+
+	"bisonai.com/orakl/node/pkg/chain/utils"
+	errorSentinel "bisonai.com/orakl/node/pkg/error"
+	"bisonai.com/orakl/node/pkg/secrets"
+	"github.com/klaytn/klaytn/common"
+	"github.com/klaytn/klaytn/crypto"
+	"github.com/rs/zerolog/log"
+)
+
+type SignerConfig struct {
+	pk string
+}
+
+type SignerOption func(*SignerConfig)
+
+func WithSignerPk(pk string) SignerOption {
+	return func(config *SignerConfig) {
+		config.pk = pk
+	}
+}
+
+func getSignerPk(ctx context.Context, config SignerConfig) (string, error) {
+	var pk string
+	var err error
+	if config.pk != "" {
+		pk = strings.TrimPrefix(config.pk, "0x")
+		return pk, nil
+	} else {
+		pk, err = utils.LoadSignerPk(ctx)
+		if err != nil || pk == "" {
+			log.Warn().Msg("failed to load signer from pgs")
+		}
+
+		if pk == "" {
+			pk = secrets.GetSecret(SignerPk)
+			if pk == "" {
+				log.Error().Msg("signer pk not set")
+				return "", errorSentinel.ErrChainSignerPKNotFound
+			}
+			err = utils.StoreSignerPk(ctx, pk)
+			if err != nil {
+				log.Warn().Msg("failed to store pk")
```

Comment:
how about including the error message:
```go
log.Warn().Err(err).Msg("failed to store pk")
```

Reply:
awh yes, will add missing logs accordingly